app.py

```diff
@@ -59,13 +60,21 @@ def generate_select_boxes(options_x, options_y, options_color, key_prefix):
         )
         select_boxes[1] = col_y
 
-    if options_color:
-        col_color = st.selectbox(
+    if options_color_1:
+        col_color_1 = st.selectbox(
             label="Color",
-            options=[""] + options_color,
-            key=f"{key_prefix}_color",
+            options=[""] + options_color_1,
+            key=f"{key_prefix}_color1",
         )
-        select_boxes[2] = col_color
+        select_boxes[2] = col_color_1
+
+    if options_color_2:
+        col_color_2 = st.selectbox(
+            label="Second Color",
+            options=[""] + options_color_2,
+            key=f"{key_prefix}_color2",
+        )
```

Comment:
Referring to my previous comment, you could call this just before `plot_donut_complex`